src/metabase/api/user.clj

```diff
@@ -39,18 +41,25 @@
 (defn- fetch-user [& query-criteria]
   (apply db/select-one (vec (cons User user/admin-or-self-visible-columns)) query-criteria))
 
-(defn- maybe-set-user-permissions-groups! [user-or-id new-groups-or-ids & [is-superuser?]]
-  ;; if someone passed in both `:is_superuser` and `:group_ids`, make sure the whether the admin group is in group_ids
-  ;; agrees with is_superuser -- don't want to have ambiguous behavior
-  (when (and (some? is-superuser?)
-             new-groups-or-ids)
-    (api/checkp (= is-superuser? (contains? (set new-groups-or-ids) (u/the-id (perms-group/admin))))
-      "is_superuser" (tru "Value of is_superuser must correspond to presence of Admin group ID in group_ids.")))
-  (when (some? new-groups-or-ids)
-    (when-not (= (user/group-ids user-or-id)
-                 (set (map u/the-id new-groups-or-ids)))
-      (api/check-superuser)
-      (user/set-permissions-groups! user-or-id new-groups-or-ids))))
+(defn- maybe-set-user-permissions-groups! [user-or-id new-groups-or-ids]
+  (when (and (some? new-groups-or-ids)
```

Comment:
if you're just trying to check that `new-groups-or-ids` is non-nil, you can just use it directly in the conditional, without using `some?`. same with the usages of `some?` below
```suggestion
  (when (and new-groups-or-ids
```